cloudflare/resource_cloudflare_worker_kv.go

```diff
@@ -0,0 +1,127 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareWorkerKV() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareWorkersKVCreate,
+		Read:   resourceCloudflareWorkersKVRead,
+		Update: resourceCloudflareWorkersKVUpdate,
```

Comment:
I see your create + update methods are pretty similar. In some resources where this is also the case, the `Create` and `Update` just share the one method. Up to you as I don't mind either way.

Reply:
Is there any harm in calling `d.SetId` in an `Update` method?  That's the main difference between them.  I'm all for saving some code if that isn't an issue.

Reply:
These have been combined into one method.